pkg/controllers/bundle/controller.go

```diff
@@ -116,10 +130,86 @@ func (h *handler) OnBundleChange(bundle *fleet.Bundle, status fleet.BundleStatus
 		return nil, status, err
 	}
 
+	if err := setResourceKey(&status, bundle, h.isNamespaced, set); err != nil {
```

Comment:
I think the set temp variable isn't needed.  Just change set here so `status.ObservedGenerate != bundle.Generation` and just always set observergeneration to generation before return.